plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -1537,4 +1553,27 @@ public int hashCode()
             return Objects.hash(baseColumnIdentity, path);
         }
     }
+
+    @NotThreadSafe
+    private static class EqualityDeleteSet
+    {
+        private final StructLikeSet deleteSet;
+        private final StructProjection projection;
+
+        public EqualityDeleteSet(Schema schema, Schema deleteSchema)
```

Comment:
the call site looks like `new EqualityDeleteSet(deleteSchema, schemaFromHandles(readColumns))`
are the constructor arg names right? is the call site right?

Reply:
I suggest  `public EqualityDeleteSet(Schema deleteSchema, Schema fileSchema)`. Because first one is schema from iceberg metadata, second schema from read file (parquet, orc, etc.)

Reply:
@Heltman yes, `public EqualityDeleteSet(Schema deleteSchema, Schema fileSchema)` should be fine. Let's go forward with this suggestion

Reply:
Finally we reached a consensus, `public EqualityDeleteSet(Schema deleteSchema, Schema dataSchema)` is good idea. `deleteSchema` come from iceberg metadata, `dataSchema` come from equality delete file.